compiler/src/codegen/mashtree.re

```diff
@@ -548,10 +556,14 @@ type mash_program = {
   exports: list(export),
   main_body: block,
   main_body_stack_size: stack_size,
-  globals: list((Ident.t, Types.allocation_type)),
+  globals: list((Ident.t, bool, Types.allocation_type, option(constant))),
```

Comment:
Can we make this a record now? It's not immediately clear what each of these are